app/dashboard/src/components/AriaComponents/Inputs/Password/Password.tsx

```diff
@@ -0,0 +1,57 @@
+/** @file A component wrapping {@link Input} with the ability to show and hide password. */
+import { useState } from 'react'
+import type { Path } from 'react-hook-form'
+
+import EyeIcon from '#/assets/eye.svg'
+import EyeCrossedIcon from '#/assets/eye_crossed.svg'
+import {
+  Button,
+  Input,
+  type FieldValues,
+  type InputProps,
+  type TSchema,
+} from '#/components/AriaComponents'
+
+// ================
+// === Password ===
+// ================
+
+/** Props for a {@link Password}. */
+export interface PasswordProps<
+  Schema extends TSchema,
+  TFieldValues extends FieldValues<Schema>,
+  TFieldName extends Path<TFieldValues>,
+  TTransformedValues extends FieldValues<Schema> | undefined = undefined,
+> extends Omit<
+    InputProps<Schema, TFieldValues, TFieldName, TTransformedValues>,
+    'addonMiddle' | 'type'
+  > {}
+
+/** A component wrapping {@link Input} with the ability to show and hide password. */
+export function Password<
+  Schema extends TSchema,
+  TFieldValues extends FieldValues<Schema>,
+  TFieldName extends Path<TFieldValues>,
+  TTransformedValues extends FieldValues<Schema> | undefined = undefined,
+>(props: PasswordProps<Schema, TFieldValues, TFieldName, TTransformedValues>) {
+  const [showPassword, setShowPassword] = useState(false)
+
+  return (
+    <Input
+      {...props}
+      type={showPassword ? 'text' : 'password'}
+      addonMiddle={
+        <Button
+          size="medium"
+          variant="icon"
+          icon={showPassword ? EyeIcon : EyeCrossedIcon}
+          className="absolute right-0 top-1/2 -translate-y-1/2"
```

Comment:
`position:absolute` here breaks visuals of the component (input shall not be under the button
![CleanShot 2024-08-06 at 18 11 15@2x](https://github.com/user-attachments/assets/b867b5be-3f8d-4283-bbd2-b3a8a7b251d0)

Reply:
true... but then i think the best way would be to do it like the old component and add the eye as an option to `Input` directly so that we get full control over styling/markup structure

Reply:
I guess using addonEnd should be sufficient?